pqarrow/arrowutils/sort.go

```diff
@@ -2,93 +2,306 @@ package arrowutils
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sort"
 
 	"github.com/apache/arrow/go/v14/arrow"
 	"github.com/apache/arrow/go/v14/arrow/array"
 	"github.com/apache/arrow/go/v14/arrow/compute"
 	"github.com/apache/arrow/go/v14/arrow/memory"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/polarsignals/frostdb/pqarrow/builder"
+)
+
+type Direction uint
+
+const (
+	Ascending Direction = iota
+	Descending
 )
 
-// SortRecord sorts the given record's rows by the given column. Currently only supports int64, string and binary columns.
-func SortRecord(mem memory.Allocator, r arrow.Record, cols []int) (*array.Int64, error) {
-	if len(cols) > 1 {
-		return nil, fmt.Errorf("sorting by multiple columns isn't implemented yet")
+func (d Direction) comparison() int {
+	switch d {
+	case Ascending:
+		return -1
+	case Descending:
+		return 1
+	default:
+		return 0
```

Comment:
I would panic in this default case since it doesn't make sense

Reply:
done c00f9d4